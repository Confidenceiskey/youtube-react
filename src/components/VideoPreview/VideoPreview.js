import React, { Component } from 'react';
import { Image } from 'semantic-ui-react';
import './VideoPreview.scss';

class VideoPreview extends Component {
  render() {
    const { video } = this.props;
    if (!video) {
      return <div />;
    }

    const horizontal = this.props.horizontal? 'horizontal' : null;
    return (
      <div className={['video-preview', horizontal].join(' ')}>
        <div className='image-container'>
          <Image 
            src={video.snippet.thumbnails.medium.url} 
            alt={`Video Preview of ${video.snippet.title}`}
          />
          <div className='time-label'>
            <span>{video.contentDetails.duration}</span>
          </div>
        </div>
        <div className='video-info'>
          <div className='semi-bold show-max-two-lines'>{video.snippet.title}</div>
          <div className='video-preview-metadata-container'>
            <div className='channel-title'>{video.snippet.channelTitle}</div>
            <div><span>{video.statistics.viewCount} Views â€¢ {video.snippet.publishedAt}</span></div>
          </div>
        </div>
      </div>
    );
  }
}

export default VideoPreview;
