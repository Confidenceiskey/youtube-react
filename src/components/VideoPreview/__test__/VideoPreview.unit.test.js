import React from 'react';
import { shallow } from 'enzyme';
import VideoPreview from '../VideoPreview';

describe('VideoPreview', () => {
  it('renders vertically', () => {
    const wrapper = shallow(
      <VideoPreview />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders horizontally', () => {
    const wrapper = shallow(
      <VideoPreview horizontally={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
