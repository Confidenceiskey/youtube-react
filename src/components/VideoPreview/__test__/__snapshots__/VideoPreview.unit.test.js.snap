// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPreview renders VideoPreview 1`] = `
<div
  className="video-preview"
>
  <div
    className="image-container"
  >
    <Image
      alt="Video Preview"
      as="img"
      src="http://via.placeholder.com/210x118"
      ui={true}
    />
    <div
      className="time-label"
    >
      <span>
        05:22
      </span>
    </div>
  </div>
  <div
    className="video-info"
  >
    <div
      className="semi-bold show-max-two-lines"
    >
      Video Title
    </div>
    <div
      className="video-preview-metadata-container"
    >
      <div
        className="channel-title"
      >
        Channel Title
      </div>
      <div>
        <span>
          2.1M Views • 2 days ago
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`VideoPreview renders horizontally 1`] = `
<div
  className="video-preview "
>
  <div
    className="image-container"
  >
    <Image
      alt="Video Preview"
      as="img"
      src="http://via.placeholder.com/210x118"
      ui={true}
    />
    <div
      className="time-label"
    >
      <span>
        05:22
      </span>
    </div>
  </div>
  <div
    className="video-info"
  >
    <div
      className="semi-bold show-max-two-lines"
    >
      Video Title
    </div>
    <div
      className="video-preview-metadata-container"
    >
      <div
        className="channel-title"
      >
        Channel Title
      </div>
      <div>
        <span>
          2.1M Views • 2 days ago
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`VideoPreview renders vertically 1`] = `
<div
  className="video-preview "
>
  <div
    className="image-container"
  >
    <Image
      alt="Video Preview"
      as="img"
      src="http://via.placeholder.com/210x118"
      ui={true}
    />
    <div
      className="time-label"
    >
      <span>
        05:22
      </span>
    </div>
  </div>
  <div
    className="video-info"
  >
    <div
      className="semi-bold show-max-two-lines"
    >
      Video Title
    </div>
    <div
      className="video-preview-metadata-container"
    >
      <div
        className="channel-title"
      >
        Channel Title
      </div>
      <div>
        <span>
          2.1M Views • 2 days ago
        </span>
      </div>
    </div>
  </div>
</div>
`;
